linode/manager
# refactor: [M3-7471] - Tabs v7 Stories

## Description 📝
This PR reorganizes @reach/tabs components around the /Tabs component directory.
Besides the v7 story, it is meant to pave the way for M3-6705 will will replace @reach/tabs with MUI tabs.
This new org with updated imports and testing will facilitate this migration. 

## Changes  🔄
- Deprecate `.mdx` with a v7 `.tsx` story
- Reorganize all Tabs components in the same directory
- Update all imports outside of directory so they reference `Tabs` and not `ReachTabs`
- Add testing for Tabs components

![Screenshot 2023-11-29 at 11 28 23](https://github.com/linode/manager/assets/130582365/1226bd3a-17c1-4171-9589-7a23d121c639)


## Preview 📷
**There should be no regressions, behavioral or visual change or as a result of this PR**

## How to test 🧪

### Verification steps 
👉 Pull local code and run `yarn up`
- Verify behavior of tabs through the app

👉 run `yarn storybook`
- Verify new story at `http://localhost:6006/?path=/docs/components-tabs--documentation`

👉 Run new unit tests
- `yarn test SafeTabPanel.test.tsx`
- `yarn test Tab.test.tsx`
- `yarn test TabLinkList.test.tsx`
- `yarn test TabList.test.tsx`

## As an Author I have considered 🤔

*Check all that apply*
